src/main/java/cubicchunks/asm/mixin/core/common/MixinWorld_Tick.java

```diff
@@ -67,18 +67,10 @@ private boolean canUpdateEntity(World _this, int startBlockX, int oldStartBlockY
 		if (!this.isCubicWorld()) {
 			return isAreaLoaded(startBlockX, oldStartBlockY, startBlockZ, endBlockX, oldEndBlockY, endBlockZ, allowEmpty);
 		}
-		//startBlockX == entityPosX - checkRadius <==> checkRadius == entityPosX - startBlockX
-		int checkRadius = updateEntity_entityPosX - startBlockX;
-		//calculate Y range
-		int startBlockY = updateEntity_entityPosY - checkRadius;
-		int endBlockY = updateEntity_entityPosY + checkRadius;
-		//if entity is in "invalid area" (outside of the world) we need to keep ticking it.
-		//otherwise entities would no longer be updated after falling out of the world
-		BlockPos entityPos = new BlockPos(updateEntity_entityPosX, updateEntity_entityPosY, updateEntity_entityPosZ);
-		if (!isValid(entityPos)) {
-			return true;
-		}
-		return isAreaLoaded(startBlockX, startBlockY, startBlockZ, endBlockX, endBlockY, endBlockZ, allowEmpty);
+		return this.isRemote() ||
+				this.getCubeFromBlockCoords(
+						new BlockPos(updateEntity_entityPosX, updateEntity_entityPosY, updateEntity_entityPosZ))
+						.getTickets().shouldTick();
```

Comment:
This is broken and less readable now.

And it loads cubes. It should NEVER load cubes.

First: it doesn't tick entities when they are outside of world border - once you fall out of the world you are stuck there forever.

Second - leave the 32 block thing. Entities can access nearby blocks, sometimes a lot of them like the ender dragon. They need some buffer radius.